src/toil/batchSystems/slurm.py

```diff
@@ -68,7 +68,25 @@ def prepareSubmission(self,
 
         def submitJob(self, subLine):
             try:
-                output = call_command(subLine)
+                # Slurm is not quite clever enough to follow the XDG spec on
+                # its own. If the submission command sees e.g. XDG_RUNTIME_DIR
+                # in our environment, it will send it along (especially with
+                # --export=ALL), even though it makes a promise to the job that
+                # Slurm isn't going to keep. It also has a tendency to create
+                # /run/user/<uid> *at the start* of a job, but *not* keep it
+                # around for the duration of the job.
```

Comment:
Is there a particular site where this was seen? Perhaps they use `pam_slurm_adopt` and didn't see

https://github.com/SchedMD/slurm/blob/de20cb6d0c737af2f6335711a603d6e62cc0d96c/doc/html/pam_slurm_adopt.shtml#L70-L76

Reply:
We're seeing this on our new "Phoenix" Slurm cluster, and I think it explains some of the errors that #4395 is meant to fix, so I suspect it is at the reporting site there as well.

I don't think we're using `pam_slurm_adopt`, though we might be. I'm also not quite sure that we have exactly the problem the note you linked is about. Maybe adding the `loginctl enable-linger` command would solve it, but the problem isn't that `/run/user/<uid>` is getting removed, or that the XDG variables aren't set. The problem is that the XDG variables *are* set, and *also* the `/run/user/<uid>` directory is getting removed, and it is the *combination* that breaks Toil. The note suggests that you need to do some special stuff to get an XDG session, but doesn't say anything about being handed a present-but-broken XDG session otherwise.

It's possible that our admins have something wrong in the Slurm configuration to cause this, but I don't think we're using any particularly fancy non-default settings for anything. So it's either misconfigured out of the box or else very easy to misconfigure on accident.